lib/matplotlib/patheffects.py

```diff
@@ -373,3 +375,168 @@ def draw_path(self, renderer, gc, tpath, affine, rgbFace):
         if clip_path:
             self.patch.set_clip_path(*clip_path)
         self.patch.draw(renderer)
+
+
+class TickedStroke(AbstractPathEffect):
+    """
+    A line-based PathEffect which draws a path with a ticked style.
+    This line style is frequently used to represent constraints in
+    optimization.  The ticks indicate that one side of the line is
+    invalid or to represent a closed boundary of a domain (i.e. a
+    wall or the edge of a pipe).
+
+    The spacing, length, and angle of ticks can be controlled.
+
+    This line style is sometimes referred to as a hatched line.
+
+    See also the :doc:`contour demo example
+    </gallery/lines_bars_and_markers/lines_with_ticks_demo>`.
+
+    See also the :doc:`contours in optimization example
+    </gallery/images_contours_and_fields/contours_in_optimization_demo>`.
+    """
+
+    def __init__(self, offset=(0, 0),
+                 spacing=10.0, angle=45.0, length=np.sqrt(2),
+                 flip=False, **kwargs):
+        """
+        Parameters
+        ----------
+        offset : pair of floats
+            The offset to apply to the path, in points.
+        spacing : float
+            The spacing between ticks in points.
+            Default is 10.0.
+        angle : float
+            The angle between the path and the tick in degrees.  The angle
+            is measured as if you were an ant walking along the curve, with
+            zero degrees pointing directly ahead, 90 to your left, 180
+            behind you, and 270 to your right.
+            Default is 45.0.
+        length : float
+            The length of the tick relative to spacing.
+            Recommend length=1.0 when angle=90 and length=2.0 when angle=60.
+            Default is 1.414 (sqrt(2)) best for angle=45.
+        flip : bool
+            Flag to flip the ticks to the other side of the line.
+            Default is False.
+        **kwargs
+            Extra keywords are stored and passed through to
+            :meth:`AbstractPathEffect._update_gc`.
+
+        For examples see
+            :doc:`/gallery/misc/tickedstroke_demo`.
+        """
+        super().__init__(offset)
+
+        self._spacing = spacing
+        self._angle = angle
+        self._length = length
+        self._flip = flip
+        self._gc = kwargs
```

Comment:
`self._gc` is an odd name; this is not a `GraphicsContext`.

Reply:
This is the same way this is handled in the other patheffect examples in this file.  See lines 208, 261, 323.

I don't know this song, I'm just trying to hum along in tune...